src/ORM/ResultSetFactory.php

```diff
@@ -234,7 +234,7 @@ protected function groupResult(array $row, array $data): EntityInterface|array
             $results = $results[$data['primaryAlias']];
         }
         if ($data['hydrate'] && !($results instanceof EntityInterface)) {
-            $results = new $data['entityClass']($results, $options);
+            return new $data['entityClass']($results, $options);
```

Comment:
This needs an annotation, or we can just revert this change so the normal return is correct.